### DISCUSSION TITLE
New No-YouTube Policy for Screen Captures

### DISCUSSION THREAD

**Discussion by nathanlesage** [Zettlr/Zettlr#4783] [Announcements]
Hello everyone,

you may have heard that as of recently YouTube has started an anti-ad block campaign in line with known behavior by its parent company, Google. This means that users with an enabled ad blocker are effectively locked out of the platform.

I know that many users like the ease of uploading screen captures of buggy behavior to YouTube to support a bug report and make it easier for me and other maintainers to understand the behavior and follow up more quickly. **I would like to explicitly encourage this behavior**, since oftentimes it is indeed extremely helpful and makes fixing bugs faster than with a simple description. So those of you who have done this in the past: thank you!

However, [ad blockers are known to enable a safe browsing experience](https://techcrunch.com/2022/12/22/fbi-ad-blocker/). Ads often contain scams, clickbait, and may even provide vectors for malicious actors to do damage to your device. Therefore, discouraging ad blockers puts users at risks. I cannot safely tell users to disable their ad blockers simply to watch a ten-second clip of some buggy behavior. If you use ad blockers: please continue to use them. If you don't, consider to begin using one.

But there is a second reason that I recently found that speaks against uploading bug videos to YouTube: Those will show up when people search for the term "Zettlr". This means that – even after the bug has been fixed and Zettlr works as intended again – these videos remain visible to potential users and can discourage them from trying out the app based on outdated and therefore wrong impressions.

Therefore, **I hereby institute a no-YouTube policy. Do not upload bug report videos to YouTube or other videos that are not intended to be shared on YouTube as a primary platform.** Videos that discuss features of Zettlr, offer help, or in any other way can be regarded as "regular" YouTube content are of course exempt from this. I only intend to discourage using YouTube as a cheap video storage.

## How to Properly Share Screen Captures of Bug Reports

You can share screen captures of buggy behavior directly on GitHub. Since 2021, the platform allows uploads of videos [as described in their blog here.](https://github.blog/2021-05-13-video-uploads-available-github/#help-a-maintainer-reproduce-a-bug) Please upload screen captures for bug reports directly on GitHub and do not use YouTube.

I have already instructed `boring-cyborg` to add a notice for new contributors, but those who have already created issues will not receive this comment, so please heed this advice. I will remind people not to use YouTube for such videos and ask people to remove them. I do not assume ill thought, so do not worry. However, if you do this repeatedly even after me pointing this out, I reserve the right to possibly take more serious steps.

> If you have uploaded any of these short bug report videos to YouTube in the past and read this, please either remove these videos (preferably) or set them to "unlisted" (if removing is not possible). You can upload those videos directly to the corresponding issue by editing it, if you want to retain the video for archival purposes.

**Comment by FeralFlora** [comment]
Nice and reasonable policy. If it's okay, I would like to recommend ShareX for screen capturing, which I've used a lot to document issues. It's FOSS, and highly configurable.

**Reply by nathanlesage** [reply]
I am absolutely not against other ways of sharing videos!

Although I think uploading those small videos that don't have any value besides the issue right within the issue itself keeps them in context and prevents them from accidentally appearing in other places without the necessary context. I mean, for most people that should be the easiest and most comfortable, especially since you are already writing in the comment form at that point.

But as far as I can see from one quick search, ShareX appears to be suitable as well, so I think it is totally fine to use that one, if you prefer.

**Reply by FeralFlora** [reply]
Oh, with ShareX, I would still upload the file directly to the issue. You _can_ upload to various media hosts, but that's not what I was suggesting, to be clear. I only recommended it for it's capturing abilities.

**Comment by nhan000** [comment]
I thought (or at least from the videos I've seen) most people upload and set the videos as “Unlisted” and thus should not show up when people search for Zettlr on YouTube.

Uploading straight to GitHub is actually much easier with the caveat is the size limit of the video. I found myself have to resort to YouTube only because of this limit.

I guess the solution for non-YouTube policy, when over size limit of GitHub, can be to use the cloud storage of choice and share the link publicly when filing an issue.

**Reply by nathanlesage** [reply]
> I thought (or at least from the videos I've seen) most people upload and set the videos as “Unlisted” and thus should not show up when people search for Zettlr on YouTube.

If you search YouTube for Zettlr, depending on which results it gives you back there are actually quite a few that are set to public. I mean, to be fair, "unlisted" is not considered the default choice for new videos, so I kind of understand why this can happen. But this is why I have also added the possibility to simply set already uploaded videos to "unlisted" if this hasn't already been done!

**Comment by sensologica** [comment]
I highly recommend the tool [LICEcap](https://www.cockos.com/licecap/). It can record any part of your screen and outputs a GIF. More compact than a video and perfect for capturing buggy behavior. It's open source, too! 👌

Example:
![example](https://github.com/Zettlr/Zettlr/assets/119978540/d1b7eec8-4505-4c78-9ed4-153291242325)
